packager/signer/signer_test.go

```diff
@@ -186,6 +202,52 @@ func (this *SignerSuite) TestSimple() {
 	this.Assert().Equal(append(payloadPrefix.Bytes(), transformedBody...), exchange.Payload)
 }
 
+func (this *SignerSuite) TestFetchSignWithForwardedRequestHeaders() {
+	urlSets := []util.URLSet{{
+		Sign:  &util.URLPattern{[]string{"https"}, "", this.certSubjectCN(), stringPtr("/amp/.*"), []string{}, stringPtr(""), false, 2000, nil},
+		Fetch: &util.URLPattern{[]string{"http"}, "", this.httpHost(), stringPtr("/amp/.*"), []string{}, stringPtr(""), false, 2000, boolPtr(true)},
+	}}
+	this.fakeHandler = func(resp http.ResponseWriter, req *http.Request) {
+		// Host and X-Foo headers are forwarded with pkgt.ForwardedRequestHeaders
+		this.Assert().Equal("www.example.com", req.Host)
+		this.Assert().Equal("foo", req.Header.Get("X-Foo"))
+		this.Assert().Equal("", req.Header.Get("X-Bar"))
+		this.lastRequest = req
+		resp.Header().Set("Content-Type", "text/html")
+		resp.Write(fakeBody)
+	}
+	// Request with ForwardedRequestHeaders
+	resp := this.getFRH(this.T(), this.new(urlSets),
+		"/priv/doc?fetch="+url.QueryEscape(this.httpURL()+fakePath)+
+			"&sign="+url.QueryEscape(this.httpSignURL_CertSubjectCN()+fakePath), "www.example.com")
+	this.Assert().Equal(fakePath, this.lastRequest.URL.String())
```

Comment:
As I made the test so as to have different sign and fetch url, `RequestURI`, `validity-url` and `cert-url` are changed and others are not. I think that the assertions are needed to be remained to check the case where sign url is different from fetch url.

Reply:
Ah, I see. Perhaps this would be better expressed as two tests, then. One for differing domain (without `forwardedRequestHeaders`), and one for just the forwarded request headers. Then again, given the current verbosity of the test harness, it's a trade-off.